### DISCUSSION TITLE
i-text font family render vertically middle

### DISCUSSION THREAD

**Discussion by RathodAnnu13** [fabricjs/fabric.js#9440] [Q&A]
I need to convert Fabric JS text into HTML.
All fontfamily  always  render Vertical-Middle.
But when I create HTML from JSON, Not all font-family render at vertically-middle.

Please  explain me how fabricJS rendering font. 

Canvas Output : 
![chokoPlainFont_canvas](https://github.com/fabricjs/fabric.js/assets/60641656/897cda02-6320-48fa-af83-b07128402bcb)

HTML Output : 
![chokoPlainFont](https://github.com/fabricjs/fabric.js/assets/60641656/a361764e-d7d0-4385-9400-48421613d1b8)

**Comment by asturur** [comment]
Isn't SVG a viable solution?

**Reply by RathodAnnu13** [reply]
I am seeking a solution to generate a **vector PDF** from HTML content, and it is imperative that SVG format not be used for this purpose

**Reply by RathodAnnu13** [reply]
Hello @asturur,

I sincerely appreciate your support. Currently, I am in the process of generating a vector PDF from a Fabric Canvas. To achieve this, I am rendering HTML from a Fabric Canvas JSON and using Puppeteer to create the PDF. However, I am encountering an issue related to the Text Font Family.
Could you kindly suggest a more effective approach to overcome this problem?

**Comment by ShaMan123** [comment]
I had a project that needed that also, it was a pain getting it to work and was an ugly workaround and didn't cover 100% of cases
I don't remember what I did there but I remember I had to re-position text in the browser after it loaded.
Text is not accessible for devs unfortunately, it is not well designed.
The only help I can offer is for you to dig into the svg export methods and try to mirror them into CSS.

**Reply by RathodAnnu13** [reply]
Thank you, @ShaMan123, for your response.
 I attempted to reposition text after the webpage loaded, but regrettably, it did not yield the desired results. 
Could you please provide an example of SVG export methods and guide me on how to achieve similar functionality using CSS?

**Comment by ShaMan123** [comment]
If you wish to generate PDFs and you can use node to do that then the easiet solution is #9185
If not look around https://github.com/fabricjs/fabric.js/blob/489ca7d53cd69e9be24a41e771e2a0b78ac362c9/src/shapes/Text/TextSVGExportMixin.ts#L145-L160

**Reply by RathodAnnu13** [reply]
Thank you @ShaMan123  for your answer; I appreciate your help. 
It didn't quite resolve my issue, but I'm grateful for your effort. 
Could you please provide more details or suggest an alternative approach to help me further?

**Reply by ShaMan123** [reply]
You didn't say if you can use node or not, I assume you can because of puppeteer so #9185 is the only thing you need, see the [In Action link](https://codesandbox.io/p/sandbox/fabric-node-sandbox-forked-3rp33s?file=/src/index.mjs)

**Reply by RathodAnnu13** [reply]
Thank you, @ShaMan123, for your response. 
I appreciate your help. 
I want a vector PDF. This solution would not work for me.

**Reply by ShaMan123** [reply]
Why wouldn't it work? The only reason I see is wanting the solution to be web based. Please be explicit.

**Reply by ShaMan123** [reply]
Also I didn't try to run node canvas in a web worker... could that work?

**Reply by RathodAnnu13** [reply]
@ShaMan123, you are indeed correct. I am seeking a web-based solution, and it is possible to execute Node Canvas within a web worker environment.

**Reply by ShaMan123** [reply]
>  it is possible to execute Node Canvas within a web worker environment.

so do it.

**Comment by lh-sn** [comment]
@RathodAnnu13 have you tried this option ?
https://github.com/alafr/SVG-to-PDFKit

**Reply by ShaMan123** [reply]
I haven't but I did use PDFKit, see this demo http://fabricjs.com/import-pdf

**Reply by ShaMan123** [reply]
Once you have something working I would appreciate you creating a demo that we could add to the new website

**Reply by RathodAnnu13** [reply]
@lh-sn  Thank you very much for your excellent response. 
Everything is functioning as anticipated. However, I have encountered an issue with PDFKit not loading server images.
Because It is using fs.readFile(src)

**Reply by lh-sn** [reply]
You can download the assets first. You may need to update image URL too, based on your downloads directory.
Haven't tried, but, pretty sure it works.
